src/lib.rs

```diff
@@ -1,90 +1,112 @@
-use display_interface::DisplayError;
+use core::marker::PhantomData;
 
-use smart_leds::{SmartLedsWrite, brightness, hsv::RGB8};
+use embedded_graphics_core::{
+    draw_target::DrawTarget,
+    geometry::{OriginDimensions, Point, Size},
+    pixelcolor::{Rgb888, RgbColor},
+    Pixel,
+};
+
+use smart_leds::{brightness, hsv::RGB8, SmartLedsWrite};
 
 struct Content<const W: usize, const H: usize>(pub [[RGB8; W]; H]);
 
-impl <const W: usize, const H: usize> Content<W, H> {
+impl<const W: usize, const H: usize> Content<W, H> {
     /// Return a slice that aliases the same memory.
     pub fn as_slice(&self) -> &[RGB8] {
         // NOTE(unsafe): Creates a shared reference to the same underlying data,
         // NOTE(unsafe): which we know is tightly packed and so we can compute how many RGB8 pixel is in there.
-        unsafe { core::slice::from_raw_parts(self as *const _ as *const RGB8,
-                                             core::mem::size_of::<Self>() / core::mem::size_of::<RGB8>()) }
+        unsafe {
+            core::slice::from_raw_parts(
+                self as *const _ as *const RGB8,
+                core::mem::size_of::<Self>() / core::mem::size_of::<RGB8>(),
+            )
+        }
     }
 }
 
 /// The wrapper for the LED driver.
 ///
 /// This receives the `SmartLedsWriter` trait implementations along with a
-/// `MatrixType` that describes the size and the pixels mapping between the LED
+/// `Transformation` that describes the pixels mapping between the LED
 /// strip placement and the matrix's x y coordinates.
-pub struct SmartLedMatrix<T, M: MatrixType, const W: usize, const H: usize> {
+pub struct SmartLedMatrix<T, M: Transformation<W, H>, const W: usize, const H: usize> {
     writer: T,
     content: Content<W, H>,
-    matrix_type: M,
+    transformation: PhantomData<M>,
     brightness: u8,
 }
 
-impl<T, M: MatrixType, const W: usize, const H: usize> SmartLedMatrix<T, M, W, H> {
+impl<T, M: Transformation<W, H>, const W: usize, const H: usize> SmartLedMatrix<T, M, W, H> {
     pub fn set_brightness(&mut self, new_brightness: u8) {
         self.brightness = new_brightness;
     }
+    pub fn get_brightness(&mut self) -> u8 {
+        self.brightness
+    }
 }
 
-impl<T: SmartLedsWrite, M: MatrixType, const W: usize, const H: usize> OriginDimensions for SmartLedMatrix<T, M, W, H> {
+impl<T: SmartLedsWrite, M: Transformation<W, H>, const W: usize, const H: usize> OriginDimensions
+    for SmartLedMatrix<T, M, W, H>
+{
     fn size(&self) -> Size {
-        self.matrix_type.size()
+        Size::new(W as u32, H as u32)
     }
 }
 
-impl<T: SmartLedsWrite, M: MatrixType, const W: usize, const H: usize> SmartLedMatrix<T, M, W, H> 
-where <T as SmartLedsWrite>::Color: From<RGB8> {
-    pub fn new(writer: T, matrix_type: M) -> Self {
+#[derive(Debug)]
+pub enum MatrixError {BusWriteError}
```

Comment:
Just out of curiosity, why did you choose to keep you own error type instead of returning `SmartLedsWrite::Error`? For future proofing?

Reply:
not really :) somehow I understood you suggest this, now removed